### PR TITLE
Add method to turn on/off CUO's follow behavior

### PR DESCRIPTION
**Pull request by sayotte**
Have verified in-game that this works, as called by the Python API.

### CODE REVIEW

File: Razor/RazorEnhanced/CUO.cs [hunk 1]
```diff
@@ -863,6 +863,96 @@ public static void PlayMacro(string macroName)
             return;
         }
 
+        private static (object, FieldInfo, FieldInfo) getFollowProps()
+        {
+            // FIXME this seems important but VisualStudio is throwing errors
```
Comment (sayotte) [inline Razor/RazorEnhanced/CUO.cs]: I'd genuinely like to get this fixed before any merging happens.

Comment (sayotte) [reply Razor/RazorEnhanced/CUO.cs]: This has been fixed.

File: Razor/RazorEnhanced/CUO.cs [hunk 2]
```diff
@@ -863,6 +863,118 @@ public static void PlayMacro(string macroName)
...
+            {
+                if (prop.Name == "Scene")
+                {
+                    piScene = prop;
+                    break;
+                }
+            }
+            if (piScene == null)
+            {
+                throw new Exception("No such property 'Scenes' of Game object");
```
Comment (credzba) [inline Razor/RazorEnhanced/CUO.cs]: maybe not a throw, how about a message and return nulls as you did if it is OSI client

File: Razor/RazorEnhanced/CUO.cs [hunk 3]
```diff
@@ -863,6 +863,118 @@ public static void PlayMacro(string macroName)
...
+            }
+            var scene = piScene.GetValue(game);
+            if (scene == null)
+            {
+                throw new Exception("Scenes property of Game object is null");
```
Comment (credzba) [inline Razor/RazorEnhanced/CUO.cs]: same comment. throw would crash the script,

**Review by credzba — CHANGES_REQUESTED** [review]
I didn't test it, but I'd avoid throw as it will crash the python script.

File: Razor/RazorEnhanced/CUO.cs (hunk 2 above)
Comment (sayotte) [reply Razor/RazorEnhanced/CUO.cs]: @credzba if this breaks, it means the targeted code over in CUO has changed and this API is now broken.

You're saying it's better to print something and proceed with unexpected behavior, than to break and force it to be addressed?

**Comment by cesare-montresor** [comment]
Absolutely! To exception should ever reach the user. Instead a message
explaining the issues.
People never put try catch in the python scripts

Il sab 23 mar 2024, 14:12 sayotte ***@***.***> ha scritto:

File: Razor/RazorEnhanced/CUO.cs (hunk 2 above)
Comment (sayotte) [reply Razor/RazorEnhanced/CUO.cs]: @credzba I've pushed a commit addressing the exception-throwing.

**Comment by lyncher48** [comment]
I can't wait this!!!

**Review by credzba — APPROVED** [review]

**Pull request merged**